modules/skybrowser/src/wwtdatahandler.cpp

```diff
@@ -0,0 +1,375 @@
+    XMLElement* root = doc->RootElement();
+    XMLElement* element = root->FirstChildElement(wwt::Folder.c_str());
+    const bool folderExists = element;
+    const bool folderContainNoUrls = folderExists && !hasAttribute(element, wwt::Url);
+        
+    // If the file contains no folders, or there are folders but without urls, 
+    // stop recursion
+    if (!folderExists || folderContainNoUrls) {
+        _xmls.push_back(doc);
+        LINFO("Saving " + url);
+
+        return true;
+    }
+
+    // Iterate through all the folders in the XML file 
+    while (element && std::string(element->Value()) == wwt::Folder) {
+
+        // If folder contains urls, download and parse those urls
+        if (hasAttribute(element, wwt::Url) && hasAttribute(element, wwt::Name)) {
+            std::string url = attribute(element, wwt::Url);
+            std::string fileName = attribute(element, wwt::Name);
+            downloadAndParseWtmlFilesFromUrl(_xmls, directory, url, fileName);
+        }
+        element = element->NextSiblingElement();
+    }
+    return true;
+}
+
+WwtDataHandler::~WwtDataHandler() {
+    // Call destructor of all allocated xmls
+    _xmls.clear();
+}
+
+void WwtDataHandler::loadImages(const std::string& root, 
+                                const std::filesystem::path& directory) {
+
+    // Collect the wtml files, either by reading from disc or from a url
+    if (directoryExists(directory)) {
+        parseWtmlsFromDisc(_xmls, directory);
+        LINFO("Loading images from directory");
+    }      
+    else {    
+        downloadAndParseWtmlFilesFromUrl(_xmls, directory, root, "root");
+        LINFO("Loading images from url");
+    }
+
+    // Traverse through the collected wtml documents and collect the images
+    for (tinyxml2::XMLDocument* doc : _xmls) {
+        tinyxml2::XMLElement* root = doc->FirstChildElement();
+        std::string collectionName = attribute(root, wwt::Name);
+        saveImagesFromXml(root, collectionName);
+    }
+
+    // Sort images in alphabetical order
+    std::sort(_images.begin(), _images.end(), [](ImageData& a, ImageData& b) {
+        // If the first character in the names are lowercase, make it upper case
+        if (std::islower(a.name[0])) {
+            // convert string to upper case
+            a.name[0] = ::toupper(a.name[0]);
+        }
+        if (std::islower(b.name[0])) {
+            b.name[0] = ::toupper(b.name[0]);
+        }
+        return a.name < b.name;
+        });
+
+    LINFO("Loaded " + std::to_string(_images.size()) + " WorldWide Telescope "
+            "images.");
+}
+
+int WwtDataHandler::nLoadedImages() const
+{
+    return _images.size();
+}
+
+const ImageData& WwtDataHandler::getImage(const int i) const
+{
+    ghoul_assert(i < _images.size(), "Index outside of image vector boundaries!");
+    return _images[i];
+}
+
+void WwtDataHandler::saveImageFromNode(tinyxml2::XMLElement* node,
+    std::string collection) {
```

Comment:
align argument with argument on previous line and move bracket to next line

(and remove the extra empty line, but hopefully you get this by now 😄 Always avoid unnecessary empty lines, with unnecessary mening that they aren't needed to increase help readability)